src/main/java/org/broadinstitute/hellbender/tools/HaplotypeCallerSpark.java

```diff
@@ -234,16 +266,15 @@ protected void runTool(final JavaSparkContext ctx) {
      *
      * This will be replaced by a parallel writer similar to what's done with {@link org.broadinstitute.hellbender.engine.spark.datasources.ReadsSparkSink}
      */
-    private void writeVariants(JavaRDD<VariantContext> variants, HaplotypeCallerEngine hcEngine) {
+    private static void writeVariants(String outputFile, JavaRDD<VariantContext> variants, HaplotypeCallerEngine hcEngine, SAMSequenceDictionary referenceDictionary, SAMSequenceDictionary readsDictionary) {
```

Comment:
I think I had a comment about this in another PR but I can't find it again, so I'm sorry if this is redundant.  It seems like we probably don't need two different sequence dictionaries here.  Shouldn't we be able to use getBestAvailableSequenceDictionary() for both?  The fact that they're potentially different, seems problematic to me, probably a bug in the existing code.e

Reply:
I've changed this to just take a single reference.